ignite/contrib/handlers/throughput_benchmark_wrapper.py

```diff
@@ -0,0 +1,216 @@
+import torch
+import torch.jit
+import contextlib
+from typing import Callable, Union
+from ignite.engine import Events, Engine
+from torch.utils import ThroughputBenchmark
+from torch.utils.throughput_benchmark import ExecutionStats  # for linting
+
+
+# TODO: Discuss device implications
```

Comment:
What do you mean by device implications?

Reply:
Yeah, sorry for that cryptic todo. I was referring to my comment from my previous post.
As in: How should we handle the different devices. Should we support them? Should we assume everything always runs on CPU? Should we move everything to CPU for easier use?

Reply:
@kai-tub I'm trying to play a bit with the implementation and unfortunately device implication and evaluator as an input of attach look very misleading. For example,
```python
from torch.utils.data import DataLoader
from torchvision.datasets import FakeData
from torchvision.transforms import ToTensor
from torchvision.models.resnet import resnet101
from ignite.engine import create_supervised_evaluator

device = "cuda"
model = resnet101().to(device)
dataset = FakeData(num_classes=1000, transform=ToTensor())
dataloader = DataLoader(dataset, batch_size=32, num_workers=10, shuffle=True)
evaluator = create_supervised_evaluator(model, device=device, non_blocking=True)
# evaluator's update_function passes batches to specified device etc

throughput_benchmark = ThroughputBenchmarkWrapper(model)
max_batches = 10
with throughput_benchmark.attach(evaluator, max_batches=max_batches) as evaluator_with_benchmark:
    evaluator_with_benchmark.run(dataloader)

# however here everything crashes as model is on GPU but dataloader's batches are on CPU
```

Seems like `throughput_benchmark` should behave like an `Engine` ?
```python
device = "cuda"
model = resnet101().to(device)
dataset = FakeData(num_classes=1000, transform=ToTensor())
dataloader = DataLoader(dataset, batch_size=32, num_workers=10, shuffle=True)

throughput_benchmark = create_throughput_benchmark(model, device=device, non_blocking=True, prepare_batch=None)
max_batches = 10
stats = throughput_benchmark.run(dataloader, max_batches=max_batches)
```
What do you think ?

Reply:
Yes, I like the idea of using it more like an engine more. In my mind, the user will not run the benchmark after every iteration/epoch, as the performance shouldn't change when the weights change. 
More like a separate step of the pipeline, where the performance of different models are quickly evaluated on their own, maybe with different types of input data.